### ISSUE TITLE
Adding calendar via CalDAV on macOS results in SQL error

### ISSUE THREAD

**Issue by mp1409** [nextcloud/server#4364]
### Steps to reproduce
1. Add CalDAV account to macOS Calendar app.
2. Use File -> New Calendar to add a calendar.

### Expected behaviour
A new calendar appears on the server.

### Actual behaviour
macOS calendar returns an error message:

> The server responded with an error.

> The request for account “Accountname” failed.
> The server responded with “500” to operation CalDAVAddCalendarQueueableOperation.

There is also a SQL error in the Nextcloud log (see below).

Adding a new calendar via the web interface works fine. To get rid of the error message in macOS calendar, go to Calendar -> Preferences -> Accounts and untick and retick "Disable this account".

### Server configuration

**Operating system**: FreeBSD 11.0

**Web server:** Nginx 1.11.10 

**Database:** PostgreSQL 9.6.2

**PHP version:** 7.0.17

**Nextcloud version:** Nextcloud 11.0.2

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from Nextcloud 10

**Where did you install Nextcloud from:** FreeBSD Ports collection

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.4.1
  - admin_audit: 1.1.0
  - calendar: 1.5.2
  - contacts: 1.5.3
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - survey_client: 0.1.5
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - comments
  - encryption
  - external
  - files_accesscontrol
  - files_automatedtagging
  - files_external
  - files_retention
  - firstrunwizard
  - systemtags
  - templateeditor
  - theming
  - user_external
  - user_ldap
  - user_saml
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/usr\/local\/www\/nextcloud\/data",
        "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "11.0.2.7",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 2,
        "logtimezone": "UTC",
        "mail_smtpmode": "smtp",
        "mail_from_address": "cloud",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "mail_smtpsecure": "tls",
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "theme": ""
    },
    "apps": {
        "activity": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "2.4.1"
        },
        "admin_audit": {
            "types": "logging",
            "enabled": "yes",
            "installed_version": "1.1.0"
        },
        "announcementcenter": {
            "signed": "true",
            "installed_version": "1.1.2",
            "types": "",
            "ocsid": "173921",
            "enabled": "no"
        },
        "backgroundjob": {
            "lastjob": "10"
        },
        "calendar": {
            "types": "",
            "ocsid": "168707",
            "installed_version": "1.5.2",
            "enabled": "yes"
        },
        "comments": {
            "installed_version": "0.2",
            "types": "logging",
            "enabled": "no"
        },
        "contacts": {
            "types": "",
            "ocsid": "168708",
            "installed_version": "1.5.3",
            "enabled": "yes"
        },
        "core": {
            "installedat": "1470234258.8663",
            "remote_files": "dav\/appinfo\/v1\/webdav.php",
            "remote_webdav": "dav\/appinfo\/v1\/webdav.php",
            "remote_dav": "dav\/appinfo\/v2\/remote.php",
            "remote_contacts": "dav\/appinfo\/v1\/carddav.php",
            "remote_carddav": "dav\/appinfo\/v1\/carddav.php",
            "remote_calendar": "dav\/appinfo\/v1\/caldav.php",
            "remote_caldav": "dav\/appinfo\/v1\/caldav.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "public_files": "files_sharing\/public.php",
            "public_documents": "documents\/public.php",
            "oc.integritycheck.checker": "[]",
            "repairlegacystoragesdone": "yes",
            "backgroundjobs_mode": "cron",
            "vendor": "nextcloud",
            "lastcron": "1492429500",
            "moveavatarsdone": "yes",
            "previewsCleanedUp": "1",
            "umgmt_show_backend": "false",
            "umgmt_show_last_login": "true",
            "umgmt_show_storage_location": "false",
            "lastupdateResult": "[]",
            "lastupdatedat": "1492428998"
        },
        "dav": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.1.1"
        },
        "direct_menu": {
            "installed_version": "0.9.3",
            "types": "",
            "enabled": "no"
        },
        "documents": {
            "types": "",
            "ocsid": "168711",
            "installed_version": "0.13.1",
            "enabled": "no"
        },
        "federatedfilesharing": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.1.1"
        },
        "federation": {
            "types": "authentication",
            "enabled": "yes",
            "installed_version": "1.1.1"
        },
        "files": {
            "types": "filesystem",
            "enabled": "yes",
            "cronjob_scan_files": "500",
            "installed_version": "1.6.1"
        },
        "files_accesscontrol": {
            "installed_version": "1.1.2",
            "types": "filesystem",
            "enabled": "no"
        },
        "files_external": {
            "installed_version": "1.0.2",
            "ocsid": "166048",
            "types": "filesystem",
            "enabled": "no"
        },
        "files_pdfviewer": {
            "ocsid": "166049",
            "types": "",
            "enabled": "yes",
            "installed_version": "1.0.1"
        },
        "files_sharing": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.1.1"
        },
        "files_texteditor": {
            "ocsid": "166051",
            "types": "",
            "enabled": "yes",
            "installed_version": "2.2"
        },
        "files_trashbin": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.1.0"
        },
        "files_versions": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.4.0"
        },
        "files_videoplayer": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.0.0"
        },
        "firstrunwizard": {
            "installed_version": "1.1",
            "ocsid": "166055",
            "types": "",
            "enabled": "no"
        },
        "gallery": {
            "types": "",
            "enabled": "yes",
            "installed_version": "16.0.0"
        },
        "logreader": {
            "installed_version": "2.0.0",
            "ocsid": "170871",
            "types": "",
            "enabled": "yes",
            "relativedates": ""
        },
        "lookup_server_connector": {
            "installed_version": "1.0.0",
            "types": "authentication",
            "enabled": "yes"
        },
        "nextcloud_announcements": {
            "installed_version": "1.0",
            "types": "logging",
            "enabled": "yes",
            "pub_date": "Sat, 10 Dec 2016 00:00:00 +0100"
        },
        "notifications": {
            "types": "logging",
            "enabled": "yes",
            "installed_version": "1.0.1"
        },
        "ownbackup": {
            "installed_version": "0.3.8",
            "types": "",
            "ocsid": "170674",
            "enabled": "no"
        },
        "password_policy": {
            "types": "",
            "enabled": "yes",
            "minLength": "8",
            "installed_version": "1.1.0"
        },
        "provisioning_api": {
            "types": "prevent_group_restriction",
            "enabled": "no",
            "installed_version": "1.1.0"
        },
        "serverinfo": {
            "enabled": "yes",
            "installed_version": "1.1.1",
            "types": ""
        },
        "sharebymail": {
            "installed_version": "1.0.1",
            "types": "filesystem",
            "enabled": "yes"
        },
        "shorten": {
            "enabled": "no",
            "installed_version": "0.0.15",
            "types": "filesystem",
            "ocsid": "168591"
        },
        "survey_client": {
            "installed_version": "0.1.5",
            "types": "",
            "enabled": "yes",
            "last_sent": "1491047103",
            "last_report": "{\"id\":\"oc1tl3gtmz9v\",\"items\":[[\"server\",\"version\",\"11.0.2.7\"],[\"server\",\"code\",\"other\"],[\"server\",\"enable_avatars\",\"yes\"],[\"server\",\"enable_previews\",\"yes\"],[\"server\",\"memcache.local\",\"\\\\OC\\\\Memcache\\\\APCu\"],[\"server\",\"memcache.distributed\",\"none\"],[\"server\",\"asset-pipeline.enabled\",\"no\"],[\"server\",\"filelocking.enabled\",\"yes\"],[\"server\",\"memcache.locking\",\"none\"],[\"server\",\"debug\",\"no\"],[\"server\",\"cron\",\"cron\"],[\"php\",\"version\",\"7.0.17\"],[\"php\",\"memory_limit\",536870912],[\"php\",\"max_execution_time\",0],[\"php\",\"upload_max_filesize\",524288000],[\"database\",\"type\",\"pgsql\"],[\"database\",\"version\",\"PostgreSQL 9.6.2 on amd64-portbld-freebsd11.0, compiled by FreeBSD clang version 3.8.0 (tags\\\/RELEASE_380\\\/final 262564) (based on LLVM 3.8.0), 64-bit\"],[\"database\",\"size\",125385240],[\"apps\",\"templateeditor\",\"disabled\"],[\"apps\",\"firstrunwizard\",\"disabled\"],[\"apps\",\"comments\",\"disabled\"],[\"apps\",\"systemtags\",\"disabled\"],[\"apps\",\"nextcloud_announcements\",\"1.0\"],[\"apps\",\"files_sharing\",\"1.1.1\"],[\"apps\",\"dav\",\"1.1.1\"],[\"apps\",\"files_external\",\"disabled\"],[\"apps\",\"announcementcenter\",\"disabled\"],[\"apps\",\"files_pdfviewer\",\"1.0.1\"],[\"apps\",\"password_policy\",\"1.1.0\"],[\"apps\",\"files_trashbin\",\"1.1.0\"],[\"apps\",\"updatenotification\",\"1.1.1\"],[\"apps\",\"theming\",\"disabled\"],[\"apps\",\"federation\",\"1.1.1\"],[\"apps\",\"contacts\",\"1.5.3\"],[\"apps\",\"files\",\"1.6.1\"],[\"apps\",\"activity\",\"2.4.1\"],[\"apps\",\"files_versions\",\"1.4.0\"],[\"apps\",\"workflowengine\",\"1.1.1\"],[\"apps\",\"documents\",\"disabled\"],[\"apps\",\"serverinfo\",\"1.1.1\"],[\"apps\",\"survey_client\",\"0.1.5\"],[\"apps\",\"federatedfilesharing\",\"1.1.1\"],[\"apps\",\"ownbackup\",\"disabled\"],[\"apps\",\"files_texteditor\",\"2.2\"],[\"apps\",\"shorten\",\"disabled\"],[\"apps\",\"provisioning_api\",\"disabled\"],[\"apps\",\"files_videoplayer\",\"1.0.0\"],[\"apps\",\"gallery\",\"16.0.0\"],[\"apps\",\"admin_audit\",\"1.1.0\"],[\"apps\",\"notifications\",\"1.0.1\"],[\"apps\",\"twofactor_backupcodes\",\"1.0.0\"],[\"apps\",\"files_accesscontrol\",\"disabled\"],[\"apps\",\"lookup_server_connector\",\"1.0.0\"],[\"apps\",\"sharebymail\",\"1.0.1\"],[\"apps\",\"logreader\",\"2.0.0\"],[\"apps\",\"direct_menu\",\"disabled\"],[\"apps\",\"calendar\",\"1.5.2\"],[\"stats\",\"num_files\",49350],[\"stats\",\"num_users\",11],[\"stats\",\"num_storages\",32],[\"stats\",\"num_storages_local\",1],[\"stats\",\"num_storages_home\",11],[\"stats\",\"num_storages_other\",20],[\"stats\",\"num_comments\",0],[\"stats\",\"num_comment_markers\",0],[\"stats\",\"num_systemtags\",0],[\"stats\",\"num_systemtags_mappings\",0],[\"files_sharing\",\"num_shares\",29],[\"files_sharing\",\"num_shares_user\",14],[\"files_sharing\",\"num_shares_groups\",1],[\"files_sharing\",\"num_shares_link\",13],[\"files_sharing\",\"num_shares_link_no_password\",12],[\"files_sharing\",\"num_fed_shares_sent\",1],[\"files_sharing\",\"num_fed_shares_received\",0],[\"files_sharing\",\"permissions_0_15\",1],[\"files_sharing\",\"permissions_0_1\",1],[\"files_sharing\",\"permissions_0_17\",8],[\"files_sharing\",\"permissions_6_7\",1],[\"files_sharing\",\"permissions_1_31\",1],[\"files_sharing\",\"permissions_3_1\",12],[\"files_sharing\",\"permissions_3_15\",1],[\"files_sharing\",\"permissions_0_31\",4],[\"encryption\",\"enabled\",\"no\"],[\"encryption\",\"default_module\",\"no\"]]}"
        },
        "systemtags": {
            "installed_version": "0.2",
            "types": "logging",
            "enabled": "no"
        },
        "templateeditor": {
            "installed_version": "0.1",
            "types": "",
            "enabled": "no"
        },
        "theming": {
            "types": "logging",
            "installed_version": "1.0.1",
            "enabled": "no"
        },
        "twofactor_backupcodes": {
            "installed_version": "1.0.0",
            "types": "",
            "enabled": "yes"
        },
        "updatenotification": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.1.1",
            "contacts": "1.5.3",
            "core": "11.0.2.7",
            "calendar": "1.5.2"
        },
        "workflowengine": {
            "installed_version": "1.1.1",
            "types": "filesystem",
            "enabled": "yes"
        }
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Software:** Calendar 9.0 (2155.21)

**Operating system:** macOS Sierra 10.12.4

#### Web server error log
```
***REMOVED IP *** - ***REMOVED USERNAME *** [17/Apr/2017:13:59:34 +0200] "MKCALENDAR /remote.php/dav/calendars/***REMOVED USERNAME***/2206DD94-EC43-439D-A9DB-D9E8641C04C2/ HTTP/1.1" 500 1184 "-" "Mac+OS+X/10.12.4 (16E195) CalendarAgent/386"
```

#### Nextcloud log (data/nextcloud.log)
```
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO "oc_calendars" ("principaluri", "uri", "synctoken", "transparent", "components", "displayname", "timezone", "calendarorder", "calendarcolor") VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["principals\/users\/***REMOVED USERNAME ***", "2206DD94-EC43-439D-A9DB-D9E8641C04C2", 1, false, "VEVENT", "Untitled", "BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-\/\/Apple Inc.\/\/Mac OS X 10.12.4\/\/EN\r\nCALSCALE:GREGORIAN\r\nBEGIN:VTIMEZONE\r\nTZID:Europe\/Berlin\r\nBEGIN:DAYLIGHT\r\nTZOFFSETFROM:+0100\r\nRRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU\r\nDTSTART:19810329T020000\r\nTZNAME:GMT+2\r\nTZOFFSETTO:+0200\r\nEND:DAYLIGHT\r\nBEGIN:STANDARD\r\nTZOFFSETFROM:+0200\r\nRRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU\r\nDTSTART:19961027T030000\r\nTZNAME:GMT+1\r\nTZOFFSETTO:+0100\r\nEND:STANDARD\r\nEND:VTIMEZONE\r\nEND:VCALENDAR\r\n", "2802", "#711A76FF"]: SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer: ""
```

**Comment by mp1409** [comment]
From my understanding, this is due to the `false` added to the `transparent` column of type `smallint`. In Postgres (other than in MySQL) this is should not be autoconverted to 0, because Postgres' Booleans are distinct types.

**Comment by mp1409** [comment]
Appears to be already fixed in #4166, somehow this did not turn up in the search. Anyway, sorry for the noise.
